docs/design/security/README.adoc

```diff
@@ -0,0 +1,534 @@
+= Hazelcast Security Overview
+:lang: en
+:toc:
+
+[[intro]]
+== Intro
+
+=== Enterprise Security
+
+Hazelcast Security is a set of features in Hazelcast Enterprise with configuration in several places:
+
+* **security** - most of the security configuration (identity, authentication, authorization)
+* **network** / **advanced-network** - TLS configuration for data in-transfer encryption
+* **persistence** - data-at-rest encryption
+* **auditlog** - configuration of how auditable events will be logged
+
+=== Other security-related areas
+
+Hazelcast also has other security-related areas available to open-source users too:
+
+* network and advanced-network configuration allows specifying bind interfaces
+* advanced-network allows separating configuration for different protocols footnote:[Members are expected to run in a trusted environment.]
```

Comment:
Member protocol is missing some safety guards available in the client protocol. (see [the related wiki](https://hazelcast.atlassian.net/wiki/spaces/IMDG/pages/1189937402)).
TLS with mutual authentication enabled helps a lot, but it's only available in the Enterprise edition and introduces additional overhead.

I'll rephrase it `expected` -> `recommended`.

Reply:
I see. I would find this information very useful. I wonder if it's possible to include in the docs? Because the link you sent me is an internal document.